waspc/data/Generator/templates/react-app/src/auth/user.ts

```diff
@@ -0,0 +1,15 @@
+import type { User } from './types' 
+
+export function getEmail(user: User) {
+  return findUserIdentity(user, "email")?.providerUserId ?? null;
+}
+
+export function getUsername(user: User) {
+  return findUserIdentity(user, "username")?.providerUserId ?? null;
```

Comment:
I've added an extra helper function`getFirstProviderUserId` which returns the first `providerUserId` it can find for the user.

I'd avoid all the hacking around with compiler and type assertion for now until we get some initial user feedback about the helpers we have now which are pretty lightweight and generic.

Reply:
Makes sense @infomiho , good call on not over-engineering this while we are still not sure what we need!